### PR TITLE
Change test_run_io_multiple_pods to factory

### PR DESCRIPTION
**Pull request by ebenahar**
Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

**Comment by ebenahar** [comment]
Changed the number of pods and PVCs created for the test in which IO will be running from 10 to 5 as with 10, with no relation to this PR, the test fails on I/O error. This is a workaround and I'm investigating this failure

File: tests/manage/pv_services/test_run_io_multiple_pods.py [hunk 1]
```diff
@@ -1,103 +1,54 @@
...
 
 from ocs_ci.framework.testlib import ManageTest, tier2
-from tests.fixtures import (
-    create_rbd_storageclass, create_ceph_block_pool,
-    create_rbd_secret, create_pods, create_pvcs, create_project,
-    create_cephfs_secret, create_cephfs_storageclass
-)
 
 
 logger = logging.getLogger(__name__)
```
Comment (sidhant-agrawal) [inline tests/manage/pv_services/test_run_io_multiple_pods.py]: `logger` is not being used later in the script, I think can be removed

File: tests/manage/pv_services/test_run_io_multiple_pods.py [hunk 2]
```diff
@@ -1,103 +1,54 @@
...
-    interface = constants.CEPHFILESYSTEM
-    storage_type = 'fs'
-
-    def test_run_io_multiple_pods_fs(self):
         """
-        Run IO on multiple pods in parallel - CephFS
+        custom_pvc = pvc_factory(size=self.pvc_size)
+        pods = [
+            pod_factory(
+                interface=request.param, pvc=custom_pvc
```
Comment (sidhant-agrawal) [inline tests/manage/pv_services/test_run_io_multiple_pods.py]: If I understood correctly, Isn't this trying to attach the same pvc to all the pods, which will cause the test to fail.
Is this script passing?

File: tests/manage/pv_services/test_run_io_multiple_pods.py (hunk 1 above)
Comment (ebenahar) [reply tests/manage/pv_services/test_run_io_multiple_pods.py]: right, removed

File: tests/manage/pv_services/test_run_io_multiple_pods.py (hunk 2 above)
Comment (ebenahar) [reply tests/manage/pv_services/test_run_io_multiple_pods.py]: Great catch!

File: tests/conftest.py [hunk 3]
```diff
@@ -308,10 +308,12 @@ def finalizer():
         Delete the PVC
         """
         for instance in instances:
+            backed_pv = instance.backed_pv
             instance.delete()
             instance.ocp.wait_for_delete(
                 instance.name
             )
+            helpers.validate_pv_delete(backed_pv)
```
Comment (RazTamir) [inline tests/conftest.py]: in reclaim policy 'retain' it will fail

Comment (ebenahar) [reply tests/conftest.py]: removed, covered as part of https://github.com/red-hat-storage/ocs-ci/pull/636/

File: tests/manage/pv_services/test_run_io_multiple_pods.py [hunk 4]
```diff
@@ -1,103 +1,52 @@
...
-        Run IO on multiple pods in parallel
+    @pytest.fixture()
+    def pods(self, interface, pod_factory, multi_pvc_factory):
         """
+        Prepare multiple pods for the test
 
-        for pod in self.pod_objs:
-            pod.run_io('fs', f'{self.pvc_size_int - 1}G')
+        Returns:
+            list: Pod instances
```
Comment (jilju) [inline tests/manage/pv_services/test_run_io_multiple_pods.py]: Blank line can be removed.

Comment (ebenahar) [reply tests/manage/pv_services/test_run_io_multiple_pods.py]: No need, see https://www.sphinx-doc.org/en/1.5/ext/example_google.html

**Comment by ebenahar** [comment]
Verified:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/544/console

**Pull request merged**